atlasdb-cassandra/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolTest.java

```diff
@@ -350,34 +430,53 @@ public void hostsAreNotRemovedOrAddedWhenRefreshIsDisabled() {
 
         setCassandraServersTo(CASS_SERVER_1);
         createClientPool();
-        assertThat(poolServers).containsExactlyInAnyOrder(CASS_SERVER_1, CASS_SERVER_2);
+        assertThatMap(poolServers).containsOnlyKeys(CASS_SERVER_1, CASS_SERVER_2);
 
         setCassandraServersTo(CASS_SERVER_1, CASS_SERVER_2, CASS_SERVER_3);
-        deterministicExecutor.tick(POOL_REFRESH_INTERVAL_SECONDS, TimeUnit.SECONDS);
-        assertThat(poolServers).containsExactlyInAnyOrder(CASS_SERVER_1, CASS_SERVER_2);
+        refreshPool();
+        assertThatMap(poolServers).containsOnlyKeys(CASS_SERVER_1, CASS_SERVER_2);
+    }
+
+    @Test
+    public void throwsWhenTryingToAddServerPresentInCurerntServers() {
```

Comment:
```suggestion
    public void throwsWhenTryingToAddServerPresentInCurrentServers() {
```

Reply:
Although, I don't quite see why adding a server that's already present is a problem - shouldn't that be a no-op?

Reply:
If it can be, then we can also remove the visible for testing on the validate method (yay!)

Reply:
Yeah sorry should have left a comment for you explaining this in a bit more detail. 

Basically, it should be impossible for us to have conflicting maps, due to the nature of _how_ we determine `serversToAdd` (i.e. servers not present in the pool presently). If this sometime in the future throws, we then have strong evidence of a bug, and suggests something sketchy going on. 

That said your other comment here has a valid point (which I initially went with), which was it shouldn't really matter, so arguably we shouldn't break in production on this. However, this means we are OK with potentially introducing this as a bug in the future, which is not ideal. 

I don't feel strongly either way, but opt'd for this mode just b/c this _really_ shouldn't happen, and could introduce a nasty bug later on if we change assumptions of topology validation.